bitnami/owncloud/values.yaml

```diff
@@ -217,6 +221,23 @@ persistence:
 
     accessMode: ReadWriteOnce
     size: 8Gi
+    # extraVolumes:
+    #   - name: shared-data
+    #     persistentVolumeClaim:
+    #       claimName: shared-data-pvc
+
+    # extraVolumeMounts:
+    #   - name: shared-data
+    #     mountPath: /mnt/data/files/admin/files/shared-data
+
+extraEnv: {}
+  # - name: OWNCLOUD_ADMIN_PASSWORD
+  #   valueFrom:
+  #     secretKeyRef:
+  #       name: my-admin-password
+  #       key: password
+  # - name: OWNCLOUD_SUB_URL
+  #   value: "/mysubpath/owncloud"
 
```

Comment:
If you take a look to other charts (e.g. RabbitMQ), you'll find out that we provide different ways to add extra environment variables:

- Listing the extra environment variables to add
- Retrieving them from an existing ConfigMap
- Retrieving them from an existing Secret (when they contain credentials)

See:

- https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/values.yaml#L153
- https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/templates/statefulset.yaml#L159
- https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/README.md#adding-extra-environment-variables

Could we do exactly the same here?

Reply:
Thank you for the quick review, I'm happy to see you're receptive to the PR :)

The `extraEnvVars` as defined there are wrong, unless i'm missing something

For the templating in [statefulset.yaml](https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/templates/statefulset.yaml#L159), the value would need to be defined as
```
extraEnvVars:
  FOO: BAR
```
Doing it as an array of names/values as suggested in the README and values with the current templating results in:
```
            - name: 0
              value: "map[FOO value:BAR]"
```
I definitely agree with adding the possibility to retrieve from configmaps & secrets, and I don't mind the name change (although I've generally seen `extraEnv` not `extraEnvVars` in other charts personally).
I do however think that the adding should be just templated and added to the envs directly, rather than only accepting key-value pairs like is the templating code now. i.e. I agree with the example in the values and the README, but the templating in `statefulset.yaml` should be changed to just this imo:
```
        {{- if .Values.extraEnvVars }}
        {{- tpl (toYaml .Values.extraEnvVars) . | nindent 12 }}
        {{- end }}
```
I made a quick PR in Github that should fix it: https://github.com/bitnami/charts/pull/2867
I believe this is better because it allows users to define the envs any way they want (eg: they could do 
```
   - name: OWNCLOUD_ADMIN_PASSWORD
     valueFrom:
       secretKeyRef:
         name: my-admin-password
         key: password
```
which imo is a very nice functionality, and I don't see any point in removing it in favor of only key-value pairs.

I can then work on adding the other env variables in the next few days, hopefully by the end of next week, but I have a few other things that I need to deal with first for our projects before getting back to doing this. In the meantime, i'd be happy to continue discussing so we establish exactly what it should look like at the end, so that I can make the changes relatively quickly when i get back to it rather than too much back and forth at that time, if that's okay with you.

Reply:
We have supported both ways in the past, and I agree that using a function such as the one below is more flexible. Thanks for pointing it out!!
```
        {{- if .Values.extraEnvVars }}
        {{- tpl (toYaml .Values.extraEnvVars) . | nindent 12 }}
        {{- end }}
```

Reply:
PS: Use `extraEnvVars`, we're trying to make that our standard